homeassistant/components/neato/config_flow.py

```diff
@@ -32,19 +37,21 @@ async def async_step_user(self, user_input: dict | None = None) -> dict:
 
         return await super().async_step_user(user_input=user_input)
 
-    async def async_step_reauth(self, data) -> dict:
+    async def async_step_reauth(self, data: MappingProxyType[str, Any]) -> FlowResult:
         """Perform reauth upon migration of old entries."""
         return await self.async_step_reauth_confirm()
 
-    async def async_step_reauth_confirm(self, user_input: dict | None = None) -> dict:
+    async def async_step_reauth_confirm(
+        self, user_input: dict | None = None
```

Comment:
Please check all of the dict types in the integration and see if we can type them more detailed.
```suggestion
        self, user_input: dict[str, Any] | None = None
```